src/App.tsx

```diff
@@ -1,26 +1,272 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+/* eslint-disable @typescript-eslint/no-shadow */
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import { Todo } from './types/Todo';
+import { TodoComponent } from './components/Todo/todo.component';
+import { TodoStatus } from './components/Filter/filter.status';
+import { ErrorTypes } from './components/Errors/error';
+import * as Services from './api/todos';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorTitle, setErrorTitle] = useState<string | null>(null);
+  const [status, setStatus] = useState<TodoStatus>(TodoStatus.All);
+  const [newTodoTitle, setNewTodoTitle] = useState<string>('');
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    Services.getTodos()
+      .then(setTodos)
+      .catch(() => setErrorTitle(ErrorTypes.UnableToLoadTodos));
+  }, []);
+
+  useEffect(() => {
+    if (errorTitle !== null) {
+      setTimeout(() => setErrorTitle(null), 3000);
+    }
+  }, [errorTitle]);
+
+  useEffect(() => {
+    inputRef.current?.focus();
+  }, [todos, errorTitle]);
+
+  const handleAddTodo = (event: React.FormEvent<HTMLFormElement>) => {
```

Comment:
you should use useCallback here and in the other event handlers -> https://semaphoreci.com/blog/react-usecallback-hook#:~:text=useCallback%20is%20most%20commonly%20used,as%20event%20handlers%20and%20timers.